core/network/network_state.hpp

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_NETWORK_NETWORK_STATE_HPP
+#define KAGOME_CORE_NETWORK_NETWORK_STATE_HPP
+
+#include "network/peer_client.hpp"
+#include "network/peer_server.hpp"
+
+namespace kagome::network {
+
+  /// Stores network's peer information
+  struct NetworkState {
+    std::std::shared_ptr<PeerClient> peer_client;
```

Comment:
Should be a vector actually, or map id->api

Reply:
also
```suggestion
    std::vector<std::shared_ptr<PeerClient>> peers;
```

Reply:
Or, actually, it is better to make an interface PeerSet, which will incapsulate logic of getting peers (by id, as example), iterating, etc...

Reply:
I think just unordered map is enough for now